python/pyspark/sql/functions/builtin.py

```diff
@@ -15341,12 +15341,15 @@ def regexp_count(str: "ColumnOrName", regexp: "ColumnOrName") -> Column:
 
     Examples
     --------
+    >>> from pyspark.sql import functions as sf
     >>> df = spark.createDataFrame([("1a 2b 14m", r"\d+")], ["str", "regexp"])
-    >>> df.select(regexp_count('str', lit(r'\d+')).alias('d')).collect()
+    >>> df.select(sf.regexp_count('str', lit(r'\d+')).alias('d')).collect()
```

Comment:
let's make such changes:
1, replace `collect` with `show()`, and discard `alias('d')` to also test the default output column name;
2, also `show` the input columns by adding `'*'`;
3, also add `sf.` for other built-in PySpark functions, e.g. `lit`;

```suggestion
    >>> df.select('*', sf.regexp_count('str', sf.lit(r'\d+'))).show()
```

please also make such changes in other places.

Reply:
you might generate the new output of each example with REPL `bin/pyspark`:

<img width="1450" alt="image" src="https://github.com/user-attachments/assets/f7cc9cc4-9c7a-406d-8a4a-c5ed399f111c" />